src/transformers/trainer.py

```diff
@@ -1126,13 +1137,20 @@ def get_optimizer_cls_and_kwargs(
                 The training arguments for the training session.
 
         """
+        use_badam = "badam_" in args.optim
+        if use_badam:
+            args.optim = args.optim.replace("badam_", "")
 
         # parse args.optim_args
         optim_args = {}
+        badam_optim_args = {}
         if args.optim_args:
             for mapping in args.optim_args.replace(" ", "").split(","):
                 key, value = mapping.split("=")
-                optim_args[key] = value
+                if "badam_" in key and use_badam:
+                    badam_optim_args[key.replace("badam_", "")] = value
+                else:
+                    optim_args[key] = value
```

Comment:
We shouldn't need to do this string manipulation and have these parallel optim_args. This doesn't scale well if we want to add other optimizers which accept some of the previous optim args